sdk/formrecognizer/ai-form-recognizer/src/documentAnalysisClient.ts

```diff
@@ -415,40 +427,215 @@ export class DocumentAnalysisClient {
       );
     }
 
-    return this.createAnalysisPoller<unknown>(input, {
-      initialModelId,
+    return this.createUnifiedPoller<unknown>(
+      (abortSignal) => {
+        const [contentType, analyzeRequest] = toAnalyzeRequest(input);
+
+        return this._restClient.documentModels.analyzeDocument(initialModelId, contentType as any, {
+          ...options,
+          abortSignal,
+          analyzeRequest,
+        });
+      },
+      {
+        initialModelId,
+        options,
+        transformResult: (result) => transformResult(toAnalyzeResultFromGenerated(result)),
+      }
+    );
+  }
+
+  /**
+   * Classify a document using a custom classifier given by its ID.
+   *
+   * This method produces a long-running operation (poller) that will eventually produce an `AnalyzeResult`. This is the
+   * same type as `beginAnalyzeDocument` and `beginAnalyzeDocumentFromUrl`, but the result will only contain a small
+   * subset of its fields. Only the `documents` field and `pages` field will be populated, and only minimal page
+   * information will be returned. The `documents` field will contain information about all the identified documents and
+   * the `docType` that they were classified as.
+   *
+   * ### Example
+   *
+   * This method supports streamable request bodies ({@link FormRecognizerRequestBody}) such as Node.JS `ReadableStream`
+   * objects, browser `Blob`s, and `ArrayBuffer`s. The contents of the body will be uploaded to the service for analysis.
+   *
+   * ```typescript
+   * import * as fs from "fs";
+   *
+   * const file = fs.createReadStream("path/to/file.pdf");
+   *
+   * const poller = await client.beginClassifyDocument("<classifier ID>", file);
+   *
+   * // The result is a long-running operation (poller), which must itself be polled until the operation completes
+   * const {
+   *   pages, // pages extracted from the document, which contain only basic information for classifiers
+   *   documents // extracted documents and their types
+   * } = await poller.pollUntilDone();
+   *
+   * // We'll print the documents and their types
+   * for (const { docType } of documents) {
+   *   console.log("The type of this document is:", docType);
+   * }
+   * ```
+   *
+   * @param classifierId - the ID of the custom classifier to use for analysis
+   * @param document - the document to classify
+   * @param options - options for the classification operation
+   * @returns a long-running operation (poller) that will eventually produce an `AnalyzeResult`
+   */
+  public async beginClassifyDocument(
+    classifierId: string,
+    document: FormRecognizerRequestBody,
+    // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
```

Comment:
Sometimes I feel this lint rule is a bit annoying :cry: I wonder if that's just me or you also feel the same way.

Reply:
I think it's an okay rule in concept but in practice I often want to ignore it.